corehq/apps/accounting/tests/test_downgrades.py

```diff
@@ -0,0 +1,212 @@
+import datetime
+
+from corehq.apps.accounting import utils, tasks
+from corehq.apps.accounting.models import (
+    DomainUserHistory,
+    InvoiceCommunicationHistory,
+    CommunicationType,
+    DefaultProductPlan,
+    SoftwarePlanEdition,
+    SubscriptionType,
+    Subscription,
+    CustomerInvoiceCommunicationHistory,
+    CustomerInvoice,
+)
+from corehq.apps.accounting.tests.base_tests import BaseAccountingTest
+from corehq.apps.accounting.tests import generator
+from corehq.apps.accounting.utils.downgrade import (
+    DAYS_PAST_DUE_TO_TRIGGER_OVERDUE_NOTICE,
+    downgrade_eligible_domains,
+    DAYS_PAST_DUE_TO_TRIGGER_DOWNGRADE_WARNING,
+    DAYS_PAST_DUE_TO_TRIGGER_DOWNGRADE,
+)
```

Comment:
As just an offhand comment, seems like like `freezegun` would be useful here to be able to set up a subscription and then simulate moving through time. Looks like you found a different approach that worked well, so no specific suggestion here

Awesome to see these tests!